### PR TITLE
Adjusted TrustArc optout

### PR DESCRIPTION
**Pull request by hyebahi**
Users were unable to open the modal window to review the settings. Adjusted the function to check if the opt-out has already been executed.
Steps to test

- Go to TrustArc.com
- Wait the autoconsent to execute
- Click the Cookie preferences link at the bottom of the page

Before fix
The modal will autoclose after a few seconds
After fix
The modal will remain open

### CODE REVIEW

File: lib/eval-snippets.ts [hunk 1]
```diff
@@ -109,6 +111,7 @@ export const snippets = {
   EVAL_PRIMEBOX_0: () => !document.cookie.includes('cb-enabled=accepted'),
   EVAL_PUBTECH_0: () => document.cookie.includes('euconsent-v2') && (document.cookie.match(/.YAAAAAAAAAAA/) || document.cookie.match(/.aAAAAAAAAAAA/) || document.cookie.match(/.YAAACFgAAAAA/)) ,
   EVAL_REDDIT_0: () => document.cookie.includes('eu_cookie={%22opted%22:true%2C%22nonessential%22:false}'),
+  EVAL_SIBBO_0: () => !!window.localStorage.getItem('euconsent-v2'),
```
Comment (muodov) [inline lib/eval-snippets.ts:114]: This seems unrelated. A merge artifact?

File: lib/cmps/trustarc-frame.ts [hunk 2]
```diff
@@ -66,25 +66,37 @@ export default class TrustArcFrame extends AutoConsentCMPBase {
...
   async optOut() {
+
+    // if the user has already opted out, let's not close the window
+    if(await this.mainWorldEval('EVAL_TRUSTARC_FRAME')){
+      return true;
+    }
+
+    //When Tags are being controlled through a tag managment system, the window will not call the vendors' opt-out
+    let timeout = 3000;
+    if(await this.mainWorldEval('EVAL_TRUSTARC_FRAME_GTM')) {
```
Comment (muodov) [inline lib/cmps/trustarc-frame.ts:77]:
```suggestion
    if (await this.mainWorldEval('EVAL_TRUSTARC_FRAME_GTM')) {
```

File: lib/cmps/trustarc-frame.ts [hunk 3]
```diff
@@ -66,25 +66,37 @@ export default class TrustArcFrame extends AutoConsentCMPBase {
   }
 
   async optOut() {
+
+    // if the user has already opted out, let's not close the window
+    if(await this.mainWorldEval('EVAL_TRUSTARC_FRAME')){
```
Comment (muodov) [inline lib/cmps/trustarc-frame.ts:71]:
```suggestion
    if (await this.mainWorldEval('EVAL_TRUSTARC_FRAME')){
```

File: lib/eval-snippets.ts [hunk 4]
```diff
@@ -49,6 +49,8 @@ export const snippets = {
   },
   EVAL_ONETRUST_1: () => window.OnetrustActiveGroups.split(',').filter(s => s.length > 0).length <= 1,
   EVAL_TRUSTARC_TOP: () => window && window.truste && window.truste.eu.bindMap.prefCookie === '0',
+  EVAL_TRUSTARC_FRAME: () => window && window.QueryString && window.QueryString.preferences === '0',
```
Comment (muodov) [inline lib/eval-snippets.ts:52]: I know this is not followed everywhere, but it'd be nice to use a more meaningful name for snippets. Something like "EVAL_TRUSTART_FRAME_TEST"?

**Review by muodov — COMMENTED** [review]
Thanks @hyebahi! The change seems fine, just a couple nits from me and good to merge

**Comment by muodov** [comment]
@hyebahi hm, did you mean to close this PR?

**Comment by hyebahi** [comment]
Hi
@muodov 
Sorry, I closed it by mistake
Thanks

File: lib/eval-snippets.ts (hunk 1 above)
Comment (muodov) [reply lib/eval-snippets.ts:114]: I don't think this is used, could you remove this line?

File: lib/eval-snippets.ts [hunk 5]
```diff
@@ -49,6 +49,8 @@ export const snippets = {
   },
   EVAL_ONETRUST_1: () => window.OnetrustActiveGroups.split(',').filter(s => s.length > 0).length <= 1,
   EVAL_TRUSTARC_TOP: () => window && window.truste && window.truste.eu.bindMap.prefCookie === '0',
+  EVAL_TRUSTART_FRAME_TEST: () => window && window.QueryString && window.QueryString.preferences === '0',
```
Comment (muodov) [inline lib/eval-snippets.ts:52]:
```suggestion
  EVAL_TRUSTARC_FRAME_TEST: () => window && window.QueryString && window.QueryString.preferences === '0',
```

File: lib/cmps/trustarc-frame.ts [hunk 6]
```diff
@@ -116,4 +128,11 @@ export default class TrustArcFrame extends AutoConsentCMPBase {
 
     return true;
   }
+
+  async test() {
+    //Test JS variable to check the user's preference
+    //preferences = undefined means no consent is set, preferences = '0' means consent is set to required only 
+    await this.wait(500);
+    return await this.mainWorldEval('EVAL_TRUSTART_FRAME_TEST');
```
Comment (muodov) [inline lib/cmps/trustarc-frame.ts:136]:
```suggestion
    return await this.mainWorldEval('EVAL_TRUSTARC_FRAME_TEST');
```

File: lib/cmps/trustarc-frame.ts [hunk 7]
```diff
@@ -66,25 +66,37 @@ export default class TrustArcFrame extends AutoConsentCMPBase {
   }
 
   async optOut() {
+
+    // if the user has already opted out, let's not close the window
+    if (await this.mainWorldEval('EVAL_TRUSTART_FRAME_TEST')){
```
Comment (muodov) [inline lib/cmps/trustarc-frame.ts:71]:
```suggestion
    if (await this.mainWorldEval('EVAL_TRUSTARC_FRAME_TEST')){
```

File: lib/eval-snippets.ts (hunk 1 above)
Comment (hyebahi) [reply lib/eval-snippets.ts]: Done

**Review by hyebahi — COMMENTED** [review]
Changes applied

**Comment by muodov** [comment]
Thanks again @hyebahi !
